FireboltDotNetSdk.Tests/Integration/ExecuteTest.cs

```diff
@@ -285,19 +299,20 @@ public void SetGoodParameterTest(string setCommand)
         }
 
         [Test]
+        [Category("general")]
         public void SetWrongParameterTest()
         {
             using var conn = new FireboltConnection(USER_CONNECTION_STRING);
             conn.Open();
             DbCommand command = conn.CreateCommand();
             command.CommandText = "SET foo=bar";
-            string errorMessage = Assert.Throws<FireboltException>(() => command.ExecuteNonQuery()).Message;
-            Assert.That(errorMessage, Does.Contain("parameter foo is not allowed"));
-            Assert.That(conn.SetParamList, Is.EqualTo(new HashSet<string>()));
+            FireboltException exception = Assert.Throws<FireboltException>(() => command.ExecuteNonQuery());
+            Assert.That(exception.Response?.Trim(), Is.EqualTo(conn.InfraVersion == 1 ? "parameter foo is not allowed" : "query param not allowed: foo"));
```

Comment:
It makes sense to validate what kind of error we get because  otherwise we can get false positive. Test may "think" that it works (i.e. gets error) but this is a "wrong" error caused by the wrong test scenario or other problem. 

Yes, error message is a subject to change. This is obvious from the test that has to choose the expected value depending on the InfraVersion. It is better to check error code or something like that but  we do not have error code. So, although validating error message can make test fragile in future it makes it stronger in present.  

As a  compromise I can suggest not to check exact message but validate that message contains "not allowed" and the variable name "foo". In this case the expected result will not depend on the InfraVersion.

Reply:
> but we do not have error code. 

What do you mean? We have an error code from the backend. Do you mean for this specific case?

> I can suggest not to check exact message but validate that message contains "not allowed" and the variable name "foo".

Yeah, I think this will be an improvement.

Reply:
done